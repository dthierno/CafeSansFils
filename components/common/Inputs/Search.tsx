import React from "react";
import { View, TextInput, StyleSheet, TouchableOpacity } from "react-native";
import { Search, SlidersHorizontal } from "lucide-react-native";

import COLORS from "@/constants/Colors";
import SPACING from "@/constants/Spacing";
import TYPOGRAPHY from "@/constants/Typography";

type SearchBarProps = {
  placeholder?: string;
  onSearch: (text: string) => void;
  onFilter: () => void;
};

export default function SearchBar({
  placeholder = "Rechercher les caf√©s, les plats",
  onSearch,
  onFilter,
}: SearchBarProps) {
  return (
    <View style={styles.searchContainer}>
      <Search
        width={20}
        height={20}
        strokeWidth={3.5}
        color={COLORS.subtuleDark}
        testID="search-icon"
      />

      <TextInput
        style={[TYPOGRAPHY.body.large.medium, styles.searchInput]}
        placeholder={placeholder}
        placeholderTextColor={COLORS.subtuleDark}
        onChangeText={onSearch}
        returnKeyType="search"
      />

      <TouchableOpacity
        onPress={onFilter}
        style={styles.filterButton}
        testID="filter-icon-container"
      >
        <SlidersHorizontal
          width={20}
          height={20}
          strokeWidth={3}
          color={COLORS.subtuleDark}
          testID="filter-icon"
        />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  searchContainer: {
    borderRadius: 50,
    alignItems: "center",
    flexDirection: "row",
    paddingVertical: SPACING.md,
    paddingHorizontal: SPACING.lg,
    backgroundColor: COLORS.lightGray,
  },
  searchInput: {
    flex: 1,
    color: COLORS.black,
    marginLeft: SPACING.md,
    fontSize: 13.5,
  },
  filterButton: {
    marginLeft: SPACING.sm,
  },
});
