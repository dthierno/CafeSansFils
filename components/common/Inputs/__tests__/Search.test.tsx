import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import Search from "../Search";

describe("Search Component", () => {
  let tree: any;
  const mockOnSearch = jest.fn();
  const mockOnFilter = jest.fn();

  const defaultProps = {
    onSearch: mockOnSearch,
    onFilter: mockOnFilter,
    placeholder: "Rechercher les cafÃ©s, les plats",
  };

  beforeEach(() => {
    tree =render(<Search {...defaultProps} />);
  });

  it("renders correctly with default props", () => {
    // Verify the placeholder text in the TextInput
    expect(screen.getByPlaceholderText(defaultProps.placeholder)).toBeTruthy();

    // Verify the presence of the Search and Filter icons
    expect(screen.getAllByTestId("search-icon")).toBeTruthy();
    expect(screen.getAllByTestId("filter-icon")).toBeTruthy();
  });

  it("triggers onSearch callback when typing in the input", () => {
    const textInput = screen.getByPlaceholderText(defaultProps.placeholder);

    fireEvent.changeText(textInput, "New search query");

    // Verify that onSearch is called with the input value
    expect(mockOnSearch).toHaveBeenCalledWith("New search query");
    expect(mockOnSearch).toHaveBeenCalledTimes(1);
  });

  it("triggers onFilter callback when the filter button is pressed", () => {
    const filterButton = screen.getByTestId("filter-icon-container");

    fireEvent.press(filterButton);

    // Verify that onFilter is called
    expect(mockOnFilter).toHaveBeenCalledTimes(1);
  });

  it("renders with a custom placeholder", () => {
    const customPlaceholder = "Search here...";
    render(
      <Search
        {...defaultProps}
        placeholder={customPlaceholder}
      />
    );

    // Verify that the custom placeholder is displayed
    expect(screen.getByPlaceholderText(customPlaceholder)).toBeTruthy();
  });

    it("matches the snapshot", () => {
        expect(tree).toMatchSnapshot();
    });
});
