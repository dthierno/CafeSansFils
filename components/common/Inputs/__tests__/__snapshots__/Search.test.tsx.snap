// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F2F2F7",
      "borderRadius": 50,
      "flexDirection": "row",
      "paddingHorizontal": 20,
      "paddingVertical": 16,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={20}
    bbWidth={20}
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="#424242"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={3.5}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    testID="search-icon"
    vbHeight={24}
    vbWidth={24}
    width={20}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4282532418,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={3.5}
    >
      <RNSVGCircle
        cx="11"
        cy="11"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        r="8"
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={3.5}
        testID="search-icon"
      />
      <RNSVGPath
        d="m21 21-4.3-4.3"
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={3.5}
        testID="search-icon"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <TextInput
    onChangeText={
      [MockFunction] {
        "calls": [
          [
            "New search query",
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="Rechercher les cafÃ©s, les plats"
    placeholderTextColor="#424242"
    returnKeyType="search"
    style={
      [
        {
          "fontFamily": "Poppins-Medium",
          "fontSize": 13.5,
        },
        {
          "color": "#000000",
          "flex": 1,
          "marginLeft": 16,
        },
      ]
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginLeft": 12,
        "opacity": 1,
      }
    }
    testID="filter-icon-container"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#424242"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={3}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      testID="filter-icon"
      vbHeight={24}
      vbWidth={24}
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282532418,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth={3}
      >
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="21"
          x2="14"
          y1="4"
          y2="4"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="10"
          x2="3"
          y1="4"
          y2="4"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="21"
          x2="12"
          y1="12"
          y2="12"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="8"
          x2="3"
          y1="12"
          y2="12"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="21"
          x2="16"
          y1="20"
          y2="20"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="12"
          x2="3"
          y1="20"
          y2="20"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="14"
          x2="14"
          y1="2"
          y2="6"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="8"
          x2="8"
          y1="10"
          y2="14"
        />
        <RNSVGLine
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282532418,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={3}
          testID="filter-icon"
          x1="16"
          x2="16"
          y1="18"
          y2="22"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
